src/libraries/Native/Unix/System.Security.Cryptography.Native/opensslshim.c

```diff
@@ -74,7 +74,11 @@ static bool OpenLibrary()
     if (libssl == NULL)
     {
         // Prefer OpenSSL 1.1.x
+#ifdef __FreeBSD__
+        DlOpen(MAKELIB("11"));
+#else
         DlOpen(MAKELIB("1.1"));
+#endif
```

Comment:
It's a little weird that you've replaced the primary recommended name, left fallback names, and then added one more fallback.

I'd sort of expect either the whole list to be replaced, or both library probes to be additive... but one replace, one add is weird.

Reply:
As FreeBSD support is WIP at the moment I guess I was trying to make a targeted change to the bits that didn't work on my set-up.

Perhaps this function should just work through a list of versions in order of preference to match on the best library. It would be good if the list wasn't in compiled code but I really have no idea how many possible library versions are needed nor how often they change.

I'm happy to drop this for a better/more standard solution.

Reply:
The escape valve is the CLR_OPENSSL_VERSION_OVERRIDE environment variable.  There's not really any other config model that we can really have at this point.

I'd just expect this to be either

* All of the current things, then the FreeBSD things, unconditional (or an argument for why the FreeBSD ones come first)
* All of the current things, then the FreeBSD things, conditionally compiled for FreeBSD (possibly with the FreeBSD ones first)
* The entirety of this list replaced for FreeBSD.

so the versionOverride stuff, then

```C
#if __FreeBSD__
    if (libssl == NULL)
    {
        // OpenSSL 1.1.1 on FreeBSD x.y
        DlOpen(MAKELIB("11"));
    }

    if (libssl == NULL)
    {
         ...
    }
#else
    // all of the current guarded DlOpen(MAKELIB()) calls
#endif

    return libssl != NULL;
```

Reply:
I have removed the #ifdef entirely. I hope this makes sense. I am making the assumption that any non-FreeBSD system that fell through without a match previously will continue to do so. If you disagree, let me know and I'll fence the new code.